newIDE/app/src/CompactPropertiesEditor/index.js

```diff
@@ -0,0 +1,791 @@
+ * If the field does not have a `getValue` method, returns `null`.
+ */
+const getFieldValue = ({
+  instances,
+  field,
+  defaultValue,
+}: {|
+  instances: Instances,
+  field: ValueField | ActionButton | SectionTitle | Title,
+  defaultValue?: any,
+|}): any => {
+  if (!instances[0]) {
+    console.log(
+      'getFieldValue was called with an empty list of instances (or containing undefined). This is a bug that should be fixed'
+    );
+    return defaultValue;
+  }
+
+  const { getValue } = field;
+  if (!getValue) return null;
+
+  let value = getValue(instances[0]);
+  for (var i = 1; i < instances.length; ++i) {
+    if (value !== getValue(instances[i])) {
+      if (typeof defaultValue !== 'undefined') value = defaultValue;
+      break;
+    }
+  }
+
+  return value;
+};
+
+const getFieldLabel = ({
+  instances,
+  field,
+}: {|
+  instances: Instances,
+  field: ValueField,
+|}): any => {
+  if (!instances[0]) {
+    console.log(
+      'CompactPropertiesEditor._getFieldLabel was called with an empty list of instances (or containing undefined). This is a bug that should be fixed'
+    );
+    return field.name;
+  }
+
+  if (field.getLabel) return field.getLabel(instances[0]);
+
+  return field.name;
+};
+
+const CompactPropertiesEditor = ({
+  onInstancesModified,
+  instances,
+  schema,
+  mode,
+  renderExtraDescriptionText,
+  unsavedChanges,
+  project,
+  resourceManagementProps,
+  preventWrap,
+}: Props) => {
+  const forceUpdate = useForceUpdate();
+
+  const _onInstancesModified = React.useCallback(
+    (instances: Instances) => {
+      // This properties editor is dealing with fields that are
+      // responsible to update their state (see field.setValue).
+
+      if (unsavedChanges) unsavedChanges.triggerUnsavedChanges();
+      if (onInstancesModified) onInstancesModified(instances);
+      forceUpdate();
+    },
+    [unsavedChanges, onInstancesModified, forceUpdate]
+  );
+
+  const getFieldDescription = React.useCallback(
+    (field: ValueField): ?string => {
+      if (!instances[0]) {
+        console.log(
+          'CompactPropertiesEditor._getFieldDescription was called with an empty list of instances (or containing undefined). This is a bug that should be fixed'
+        );
+        return undefined;
+      }
+
+      const descriptions: Array<string> = [];
+      if (field.getDescription)
+        descriptions.push(field.getDescription(instances[0]));
+      if (renderExtraDescriptionText && field.getExtraDescription)
+        descriptions.push(
+          renderExtraDescriptionText(field.getExtraDescription(instances[0]))
+        );
+
+      return descriptions.join('\n') || undefined;
+    },
+    [instances, renderExtraDescriptionText]
+  );
+
+  const renderInputField = React.useCallback(
+    (field: ValueField) => {
+      if (field.name === 'PLEASE_ALSO_SHOW_EDIT_BUTTON_THANKS') return null; // This special property was used in GDevelop 4 IDE to ask for a Edit button to be shown, ignore it.
```

Comment:
to double check with @4ian but we should probably kill this field no? This feels like an unnecessary compatibility support